runtime/decimal.c

```diff
@@ -167,6 +167,22 @@ double _bal_decimal_to_float(TaggedPtr tp) {
     return strtod(dblStr, NULL);
 }
 
+TaggedPtr _bal_decimal_from_int(int64_t val) {
+    decQuad d;
+    if (INT32_MIN <= val && val <= INT32_MAX) {
+        decQuadFromInt32(&d, (int32_t)val);
+        return createDecimal(&d);
+    }
+    // The INT64_MIN(-9223372036854775808) has the maximum string length.
+    // It has 20 characters. Adding +1 for NULL terminator.
+    char intStr[21];
```

Comment:
I want the code to be clearly correct. How do I see how many digits are in `9223372036854775808`? How do I know that is -2^63?

What I want is `21` replaced by an expression that is clearly >=the number of digits needed.

For example, I know that one decimal digit can represents at least 3 binary bits.

So I know that ceiling(64/3) + 1 for sign + 1 for zero is enough.

Or you can have a macro that does sizeof on the stringification of INT64_MIN.

Reply:
Changed